llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -4132,19 +4133,21 @@ void llvm::UpgradeIntrinsicCall(CallBase *CI, Function *NewFn) {
       Value *Val = CI->getArgOperand(1);
       Rep = Builder.CreateAtomicRMW(AtomicRMWInst::FAdd, Ptr, Val, MaybeAlign(),
                                     AtomicOrdering::SequentiallyConsistent);
-    } else if (IsNVVM && (Name == "max.i" || Name == "max.ll" ||
-                          Name == "max.ui" || Name == "max.ull")) {
+    } else if (IsNVVM &&
+               (Name == "max.s" || Name == "max.i" || Name == "max.ll" ||
+                Name == "max.us" || Name == "max.ui" || Name == "max.ull")) {
```

Comment:
`.. && Name.consume_front("max.") ` would reduce the number of unnecessary comparisons on non-matching names.

Reply:
Fixed